# Lambda School Data Science Module 124

## Introduction to Bayesian Inference




### 1) In your own words, summarize the difference between Bayesian and Frequentist statistics.

Both have different approach to finding probabilities of events as well as different lines of thought. 

###For questions 2) - 8) use the following information:

Let A be the event of surviving the sinking of the Titanic and B be the event of being a 1st class passenger.

In [59]:
import pandas as pd
import numpy as np


data_url = 'https://raw.githubusercontent.com/Chelsea-Myers/Lambda-Intro/master/Titanic.csv'

df = pd.read_csv(data_url, skipinitialspace=True, header=0)

print(df.shape)
df.head()

(887, 8)


Survived  Pclass  ... Parents/Children_Aboard     Fare
0         0       3  ...                       0   7.2500
1         1       1  ...                       0  71.2833
2         1       3  ...                       0   7.9250
3         1       1  ...                       0  53.1000
4         0       3  ...                       0   8.0500

[5 rows x 8 columns]

2) Calculate P(A) = P(Surving)

 P(A) = 0.39

In [60]:
pd.crosstab(index = df['Survived'], columns = 'Survival rate', normalize='columns' )

col_0     Survival rate
Survived               
0              0.614431
1              0.385569

3) Calculate P(B) = P(Being a 1st class passenger)

P(B) = 0.24

In [62]:
pd.crosstab(df['Pclass'], columns = 'Class Odds', normalize = 'columns')


col_0   Class Odds
Pclass            
1         0.243517
2         0.207441
3         0.549042

4) Calculate P(A $\cap$ B) = P(Being a 1st class passenger and surviving)

P(A ∩ B) = 0.15

In [64]:
pd.crosstab(df['Pclass'], df['Survived'], normalize = 'index')

Survived         0         1
Pclass                      
1         0.370370  0.629630
2         0.527174  0.472826
3         0.755647  0.244353

5) Calculate P(A|B) = P(Surviving given that the person was a 1st class passenger)

P(A|B) = 0.62

6) Calculate (B|A) = P(Being a first class passenger given the person survived)

P(B|A) = 0.38

7) Show that 

$$P(A|B) = \frac{P(A \cap B)}{P(B)}$$

with the numbers you calculate above.

In [76]:
0.15 / 0.24

0.625

8) Show that 

$$P(A|B) = \frac{P(B|A)P(A)}{P(B)}$$

with the numbers you calculated above.

In [77]:
0.38 * 0.39 / 0.24

0.6175

###Use the following information to answer questions 9) - 11) 

Assume that the passengers on the Titanic are representative of all individuals traveling from Western Europe to the United States in the early 20th century.

Let's use the Titanic data sample to estimate the mean fare paid by all individuals traveling from Western Euorpe to the United States during that time.

9) Use a 95% frequentist CI to estimate the mean fare paid by individuals traveling from Western Europe to the US in the early 20th century.

Interpret your confidence interval.

In [88]:
df.shape

(887, 8)

In [90]:
from scipy import stats

stats.t.interval(0.95, df = 886,loc = df['Fare'].mean(), 
                 scale = df['Fare'].std()/(887**(1/2)))

(29.02482490325031, 35.586015457516254)

Your answer here.

With 95% confidence we can say that mean fare lies within 29.02 and 35.59 range 

10) Use a 95% Bayesian CI to determine the distribution of the fare paid by individuals traveling from Western Europe to the US in the early 20th century.  Interpret your results

In [80]:
from scipy import stats



(Mean(statistic=32.30542018038331, minmax=(29.024824903250337, 35.58601545751628)),
 Variance(statistic=2483.858450355689, minmax=(2262.747559034253, 2726.2430731483255)),
 Std_dev(statistic=49.824230556814946, minmax=(47.568346187714496, 52.213437668365835)))

With 95% confidence we can say that mean fare lies within 29.02 and 35.59 range 

#Portfolio Project Assignment

Watch the Porfolio Project websites mini-lecture: https://www.youtube.com/watch?v=gxbNCGERUjc&feature=youtu.be&ab_channel=LambdaSchool


Begin to set up your own Portfolio Project website.